pkgs/misc/vscode-extensions/cpptools/default.nix

```diff
@@ -73,20 +71,19 @@ vscode-utils.buildVscodeMarketplaceExtension rec {
     # Prevent download/install of extensions
     touch "./install.lock"
 
-    # Mono runtimes from nix package (used by generated `OpenDebugAD7`).
-    mv ./debugAdapters/bin/OpenDebugAD7 ./debugAdapters/bin/OpenDebugAD7_orig
-    cp -p "${openDebugAD7Script}" "./debugAdapters/bin/OpenDebugAD7"
-
     # Clang-format from nix package.
     mv  ./LLVM/ ./LLVM_orig
     mkdir "./LLVM/"
     find "${clang-tools}" -mindepth 1 -maxdepth 1 | xargs ln -s -t "./LLVM"
 
-    # Patching  cpptools and cpptools-srv
-    elfInterpreter="$(cat $NIX_CC/nix-support/dynamic-linker)"
-    patchelf --set-interpreter "$elfInterpreter" ./bin/cpptools
-    patchelf --set-interpreter "$elfInterpreter" ./bin/cpptools-srv
+    # Patching binaries
     chmod a+x ./bin/cpptools{-srv,}
+    chmod a+x ./debugAdapters/bin/OpenDebugAD7
+    patchelf --replace-needed liblttng-ust.so.0 liblttng-ust.so.1 ./debugAdapters/bin/libcoreclrtraceptprovider.so
```

Comment:
It might be better if we fed the required version here as a solib 1 is not backward compatible with solib 0 which might lead to crashes (I previously confirmed this looking at upstream code) if one of the incompatible symbol is used.

See:

- <https://github.com/NixOS/nixpkgs/issues/137314#issuecomment-933952651>

- <https://github.com/NixOS/nixpkgs/blob/8b172178c8407aaed506753374400c7b70df0850/pkgs/misc/vscode-extensions/python/default.nix#L23>

However, as doing so seem to be spread in multiple packages and that you validated that the extension works, we should only do this later in a separate PR fixing all derivations.